lib/guest_installation_and_configuration_base.pm

```diff
@@ -911,32 +927,87 @@ EOF
 }
 
 #Configure [guest_installation_method_options].User can still change [guest_installation_method],[guest_installation_media],[guest_build],[guest_version],[guest_version_major],
-#[guest_version_minor],[guest_installation_fine_grained] and [guest_autoconsole] by passing non-empty arguments using hash.Set [guest_installation_media] to the current major
-#and minor version if it does not match with [guest_version].
+#[guest_version_minor],[guest_installation_fine_grained] and [guest_autoconsole] by passing non-empty arguments using hash.Call config_guest_installation_media to set correct
+#installation media.
 sub config_guest_installation_method {
     my $self = shift;
 
     $self->reveal_myself;
     $self->config_guest_params(@_) if (scalar(@_) gt 0);
     if ($self->{guest_installation_method} eq 'location') {
-        $self->{guest_installation_media} =~ s/Build12345/Build$self->{guest_build}/g if ($self->{guest_build} ne 'gm');
+        $self->config_guest_installation_media;
+        $self->{guest_installation_method_options} = "--location $self->{guest_installation_media}";
+    }
+    $self->{guest_installation_method_options} = $self->{guest_installation_method_options} . ($self->{guest_installation_method_others} ne '' ? ",$self->{guest_installation_method_others}" : '') if ($self->{guest_installation_method_others} ne '');
+    $self->{guest_installation_method_options} = $self->{guest_installation_method_options} . ($self->{guest_installation_fine_grained} ne '' ? " --install $self->{guest_installation_fine_grained}" : '') if ($self->{guest_installation_fine_grained} ne '');
+    $self->{guest_installation_method_options} = $self->{guest_installation_method_options} . ($self->{guest_autoconsole} ne '' ? " --autoconsole $self->{guest_autoconsole}" : ' --noautoconsole');
+    return $self;
+}
+
+#Set [guest_installation_media] to the current major and minor version if it does not match with [guest_version].This subroutine also help mount nfs share if guest chooses to
+#or has to use iso installation media, for example oracle linux guest uses iso installation media from https://yum.oracle.com/oracle-linux-isos.html. Although this subroutine
+#can help correct installation media major and minor version if necessary, it is just auxiliary functionality and end user should always pay attendtion and use the meaningful
+#and correct guest parameters and profile.
+sub config_guest_installation_media {
+    my $self = shift;
+
+    $self->reveal_myself;
+    $self->{guest_installation_media} =~ s/Build12345/Build$self->{guest_build}/g if ($self->{guest_build} ne 'gm');
+#This is just auxiliary functionality to help correct and set correct installation media major and minor version if it mismatches with guest_version.It is not mandatory
```

Comment:
indent issue here and below line.

Reply:
CI tells me to do this way otherwise it will not let it pass. I really can not change as you suggested here.

Reply:
I think there should be workarounds to make it "correct", 99% sure, but sorry I do not have time to try for you. @okurz One improvement to be made in CI check script?

Reply:
I tried it again. No way. Keep it the current way unless CI changes.

Reply:
hm, I think a problem could be that the line is too long. Maybe wrapping at 80 characters line length would allow to properly indent